crates/core/src/yarn_fn/function_wrapping.rs

```diff
@@ -132,6 +143,13 @@ impl PartialEq for Box<dyn UntypedYarnFn + Send + Sync> {
 
 impl Eq for Box<dyn UntypedYarnFn + Send + Sync> {}
 
+#[macro_export]
+macro_rules! yarn_fn {
+    (($($param:ty),+) -> $ret:ty) => {
```

Comment:
I'd like the macro to be more evocative of the generated code. How about requiring the user to provide `FnMut`?
`yarn_fn_signature!{ FnMut(Foo, Bar) -> Baz }`

Reply:
(technically, they should pass an `fn`, but I think that would be leeking too much detail)

Reply:
But we return an `impl YarnFn` which is implemented for FnMuts but I could have an impl for a custom struct, right?